File: leetcode/977-squares_of_a_sorted_array/Solution.py

from typing import List, Tuple, Dict, TextIO

class Solution:
    # 双指针解决
    def sortedSquares(self, A: List[int]) -> List[int]:
        len_A = len(A) - 1
        i, j, k = 0, len_A, len_A
        res = [0] * (len_A + 1)
        while k >=0:
            if A[i]**2 < A[j]**2:
                res[k] = A[j]**2
                j -= 1
            else:
                res[k] = A[i]**2
                i += 1
            k -= 1
        return A

File: leetcode/977-squares_of_a_sorted_array/Solution.java
class Solution {
    public int[] sortedSquares(int[] A) {
        int lenA = A.length;
        int i = 0, j = lenA - 1;
        int[] res = new int[lenA];
        for (int k = lenA - 1; k >= 0; k--) {
            if (A[i]*A[i] > A[j]*A[j]) {
                res[k] = A[i]*A[i];
                i++;
            } else {
                res[k] = A[j]*A[j];
                j--;
            }
        }
        return res;
    }
}